_posts/2016-06-19-graphql-for-mobile.markdown

```diff
@@ -0,0 +1,99 @@
+
+[Here][hub] ( I strongly recommend pausing to open that link in a new window. Press cmd + enter to see the results )
+
+GraphQL comes with a playground for the API! It's amazing! Clay called it the ["Killer App" of GraphQL](https://medium.com/the-graphqlhub/graphiql-graphql-s-killer-app-9896242b2125#.6ht6374bq) - I'm inclined to concur. I've never had API docs this useful.
+
+{% expanded_img /images/2016-06-19-graphql-for-iOS-devs/graphiql.png Selection diagram %} 
+
+### How GraphQL Changed How We Write Native Code
+
+#### View Models
+
+Our GraphQL server is owned by the [web-practice](http://artsy.github.io/blog/2016/03/28/artsy-engineering-organization-stack/) and the mobile practice also help out occasionally. This ownership distinction is important, an API like this would normally be handled by our platform team. 
+
+Because of Metaphysics' ownership as a "front-end" product, it can contain additional information that is specific to our needs. For example, in our first example of a request to our GraphQL server we requested `id`, `additional_information`, `is_price_hidden` and `is_inquireable` - only two of these datums come from the database. The `is_` denotes a derived state in our implmentation.
+
+This is _awesome_, because previously a [lot of this logic](https://github.com/artsy/eigen/blob/dac7c80b66b600f9a45aaae6095544fe420f0bbc/Artsy/Views/Artwork/ARArtworkActionsView.m#L310-L362) existed in a [Google Doc](https://github.com/artsy/eigen/blob/dac7c80b66b600f9a45aaae6095544fe420f0bbc/Artsy/Views/Artwork/ARArtworkActionsView.m#L108-L109) which needed to be re-implmented in 3-4 clients. On the native side we would regularly find out we were out-of-date mid-release cycle and had to rush to catch up.
+
+So, what does this mean for view models? It lessens the need for them. If you can move a lot of your derived data to the server, it is presenting the logic that should be in a view model. We've not stopped writing view models, but now discussions on them includes "should this move to Metaphysics?".
+
+#### React Native
+
+We've already [shipped one full view controller](https://twitter.com/orta/status/734880605322776576) in [React Native](https://facebook.github.io/react-native/) for our flagship app, [Eigen](https://github.com/artsy/eigen/). The advantages that came from from GraphQL were a big part of the discussion around using Reacy Native.
+
+There will be longer articles on the "why" and "how" we choose to work this way. However the TLDR: the key thing that we're excited about React Native is Relay. Using Relay, our [views](https://github.com/artsy/emission/tree/2ac6e9fc0f85ca81483bcbd6c841841104f07833/lib/components/artist) can declare a fragment of the GraphQL query that [view needs](https://github.com/artsy/emission/blob/2ac6e9fc0f85ca81483bcbd6c841841104f07833/lib/components/artist/biography.js#L60-L69).
```

Comment:
Re articles on the “why” / “how”, I’m planning to write:

* Why we chose to use React Native: which will go into Relay/GraphQL as well.
* Using React Native in an existing native app: which will be about things we came across/for iOS devs to keep in mind.